core/sql/optimizer/RelSample.cpp

```diff
@@ -925,7 +925,6 @@ RelExpr *RelSample::bindNode(BindWA *bindWA)
   RelExpr* myChild = child(0);
   if (myChild->getOperatorType() == REL_SCAN &&
       (static_cast<Scan*>(myChild))->isHbaseTable() &&
-      myChild->selectionPred().entries() == 0 &&
```

Comment:
Thanks. That is an interesting observation. I had not thought of sampling and predicate evaluation as being commutative. I suppose even if sampling ratio is small and/or predicate selectivity is high, doing these two operations in either order will be statistically indistinguishable. Thanks for explaining.